test/policies/rule_policy_test.rb

```diff
@@ -3,15 +3,41 @@
 require 'test_helper'
 
 class RulePolicyTest < ActiveSupport::TestCase
-  test 'only rules within account profiles are accessible' do
-    assert_empty Pundit.policy_scope(users(:test), Rule)
+  setup do
     users(:test).account = accounts(:test)
-    Profile.create(name: 'test', ref_id: 'test',
-                   benchmark: benchmarks(:one), hosts: [hosts(:one)],
-                   account: accounts(:test), rules: [rules(:one)])
+  end
+
+  test 'disallows rules not in the current_user account' do
+    assert_empty Pundit.policy_scope(users(:test), Rule)
+    Profile.create!(name: 'test', ref_id: 'test',
+                    parent_profile: profiles(:one),
+                    benchmark: benchmarks(:one), rules: [rules(:one)])
+    assert_empty Pundit.policy_scope(users(:test), Rule)
+    assert_raises(Pundit::NotAuthorizedError) do
+      Pundit.authorize(users(:test), rules(:one), :index?)
+    end
+    assert_raises(Pundit::NotAuthorizedError) do
+      Pundit.authorize(users(:test), rules(:one), :show?)
+    end
+  end
+
+  test 'allows rules in the current_user account' do
+    assert_empty Pundit.policy_scope(users(:test), Rule)
+    Profile.create!(name: 'test', ref_id: 'test',
+                    parent_profile: profiles(:one),
+                    benchmark: benchmarks(:one), hosts: [hosts(:one)],
+                    account: accounts(:test), rules: [rules(:one)])
+    assert_includes Pundit.policy_scope(users(:test), Rule), rules(:one)
+    assert Pundit.authorize(users(:test), rules(:one), :index?)
+    assert Pundit.authorize(users(:test), rules(:one), :show?)
+  end
+
+  test 'allows rules from canonical profiles' do
+    assert_empty Pundit.policy_scope(users(:test), Rule)
+    Profile.create!(name: 'test', ref_id: 'test',
+                    benchmark: benchmarks(:one), rules: [rules(:one)])
```

Comment:
This tests the scenario you raised about canonical profiles with nil account_id. These three tests added should cover what we expect to receive. I'm also willing to add cases for unexpected cases, such as for profiles outside of what we validate (i.e. host+account exist along with parent_profile; nonexistent parent_profile means no host+acocunt; etc.). Let me know if you think of any cases like that.